data-prepper-test-common/src/main/java/org/opensearch/dataprepper/matcher/MapEquals.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.matcher;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Custom Hamcrest assertion {@link MapEquals#isEqualWithoutTimestamp(Map)} which compares two maps
+ * by ignoring {@link MapEquals#DEFAULT_TIMESTAMP_KEY_FOR_EVENT} key from maps.
+ * @since 1.3
+ */
+public class MapEquals extends TypeSafeMatcher<Map<String, Object>> {
+    private final Map<String, Object> expectedMap;
+    private final String DEFAULT_TIMESTAMP_KEY_FOR_EVENT = "@timestamp";
```

Comment:
Variables we are treating as constants should be marked as both `static` and `final`.